### DISCUSSION TITLE
canvas

### DISCUSSION THREAD

**Discussion by dkremsa** [open-webui/open-webui#6009] [Ideas]
Hi guys, does this tool allow for canvas functionality similar to what openAI introduced a few days back? At least the UI foundation so that we can build on top of it?

**Comment by Wadera** [comment]
This works with all models, as far as my experience goes: https://openwebui.com/f/atgehrhardt/artifacts :)

**Reply by justinh-rahb** [reply]
> This works with all models, as far as my experience goes: https://openwebui.com/f/atgehrhardt/artifacts :)

This function can be considered deprecated since its functionality has now been brought into WebUI proper.

**Reply by Wadera** [reply]
I guess on dev branch? Or it's available on stable and I can't find it on docker released `v0.3.32`?

**Reply by FedeCuci** [reply]
I can't seem to find the built-in artifacts functionality @justinh-rahb it referring to, but there seems to be a v2 of the artifacts function here: https://openwebui.com/f/tokyohouseparty/artifacts_v2.

**Reply by silentoplayz** [reply]
> I can't seem to find the built-in artifacts functionality @justinh-rahb it referring to, but there seems to be a v2 of the artifacts function here: https://openwebui.com/f/tokyohouseparty/artifacts_v2.

Simply ask any LLM to produce a web app or something using a combination of HTML, CSS, and JS within a chat on the desktop and any working code will be combined together to be rendered on the right-hand side in a pop up view automatically.

**Reply by Wadera** [reply]
I've found it here (can force it on right upper corner menu): 
![image](https://github.com/user-attachments/assets/353d283f-84f1-4fb2-bace-4e883ca8965f)

It's great! But it's a shame that it's used only to render HTML / CSS / JS. It would be great to have it similar to Anthropic Claude or OpenAI Canvas where it's show you for example bash script, yml file or text you are working on, and can make an edits, see all previous versions etc.

**Reply by FedeCuci** [reply]
I believe you do have the Artifacts function @Wadera, because I don't have it in mine. But yes, I agree that it shouldn't be used to render just HTML / CSS / JS.
![image](https://github.com/user-attachments/assets/0a1747cc-1a6d-490d-9787-bddd67ad4dfa)

**Reply by cjbarroso** [reply]
+1 for this. It should not be a lot of work (I can be mistaken, don't know the codebase) and the benefits would be great combined with, say, the free tier in google gemini flash or a local model if you have the hardware

**Comment by tjbck** [comment]
Note taking feature (similar to Notion/Obsidian), to be used alongside the current chat interface, will be worked on from the ground up and is on the roadmap. Stay tuned!

**Reply by cvaz1306** [reply]
How's this coming?

**Reply by pistonsfan2004** [reply]
following

**Comment by ipatman** [comment]
Maybe this could be integrated
https://github.com/langchain-ai/open-canvas

**Reply by sclausen** [reply]
Unfortunately, this is not being able to run offline, since the dependency of several external services, as well as the need to obtain an OpenAI, Anthropic and LangSmith API key.
